vault/raft.go

```diff
@@ -842,6 +1056,57 @@ func (c *Core) joinRaftSendAnswer(ctx context.Context, sealAccess *seal.Access,
 	return nil
 }
 
+// RaftBootstrap performs bootstrapping of a raft cluster if core contains a raft
+// backend. If raft is not part for the storage or HA storage backend, this
+// call results in an error.
+func (c *Core) RaftBootstrap(ctx context.Context, onInit bool) error {
+	if c.logger.IsDebug() {
```

Comment:
Just for my own edification, why is this `if` necessary? I haven't seen that before in our other logging code.

Reply:
If you grep for `if c.logger.IsDebug()` you'll see multiple usages of this pattern. It's a minor performance optimization that tells the logger to go through the logging logic only if we're at that log level. Technically, not doing this is fine as well, but this is done to short-circuit things.

Reply:
The only reason for doing that is if computing the arguments costs something, and even then I wouldn't do it except in the hottest of code paths.  It's an extremely micro optimization.

I suspect it also breaks the `vault monitor` feature that allows you to override the default log level.